src/Lib.P2P/PubSub/FloodRouter.cs

```diff
@@ -0,0 +1,307 @@
+        {
+            // Get binary representation
+            byte[] bin;
+            using (var ms = new MemoryStream())
+            {
+                Serializer.SerializeWithLengthPrefix(ms, msg, PrefixStyle.Base128);
+                bin = ms.ToArray();
+            }
+
+            return Task.WhenAll(peers.Select(p => SendAsync(bin, p, cancel)));
+        }
+
+        private async Task SendAsync(byte[] message, Peer peer, CancellationToken cancel)
+        {
+            try
+            {
+                using (var stream = await SwarmService.DialAsync(peer, ToString(), cancel).ConfigureAwait(false))
+                {
+                    await stream.WriteAsync(message, 0, message.Length, cancel).ConfigureAwait(false);
+                    await stream.FlushAsync(cancel).ConfigureAwait(false);
+                }
+
+                log.Debug($"sending message to {peer}");
+                return;
+            }
+            catch (Exception e)
+            {
+                log.Debug($"{peer} refused pubsub message.", e);
+            }
+        }
+
+#pragma warning disable VSTHRD100 // Avoid async void methods
+        /// <summary>
+        ///   Raised when a connection is established to a remote peer.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="connection"></param>
+        /// <remarks>
+        ///   Sends the hello message to the remote peer.  The message contains
+        ///   all topics that are of interest to the local peer.
+        /// </remarks>
+        private async void Swarm_ConnectionEstablished(object sender, PeerConnection connection)
+#pragma warning restore VSTHRD100 // Avoid async void methods
+        {
+            if (localTopics.Count == 0)
+                return;
+
+            try
+            {
+                var hello = new PubSubMessage
+                {
+                    Subscriptions = localTopics.Values
+                       .Select(topic => new Subscription
+                        {
+                            Subscribe = true,
+                            Topic = topic
+                        })
+                       .ToArray()
+                };
+                await SendAsync(hello, new Peer[] {connection.RemotePeer}, CancellationToken.None)
+                   .ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                log.Warn("Sending hello message failed", e);
+            }
+        }
+
+        /// <summary>
+        ///   Raised when the peer has no more connections.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="peer"></param>
+        /// <remarks>
+        ///   Removes the <paramref name="peer"/> from the
+        ///   <see cref="RemoteTopics"/>.
+        /// </remarks>
+        private void Swarm_PeerDisconnected(object sender, Peer peer) { RemoteTopics.Clear(peer); }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unused method parameter 'sender'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)